rl4co/utils/decoding.py

```diff
@@ -327,10 +330,14 @@ def step(
         logprobs, selected_action, td = self._step(
             logprobs, mask, td, action=action, **kwargs
         )
-        td.set("action", selected_action)
-        self.actions.append(selected_action)
-        self.logprobs.append(logprobs)
-        return td
+        # skip this step for improvement methods, since the action for improvement methods is finalized in its own policy
```

Comment:
[Internal comment] @cbhua @LTluttmann 
this may be needed in other occasions as well. We may want to standardize the API

Reply:
Agree, maybe we can rename this `step()` -> `_step()` and adding a `step()` wrapper outside. The `_step()` calculates the `logprobs, selected_actions` and `step()` selects variables to return.

Reply:
Yeah, I agree with what @cbhua suggested. We can do it! Currently, I have not performed any refactoring here for now.

Reply:
another option might be to define another non-internal function like `select_action()` which only returns `logprobs, selected_actions`. And the `step()` method additionally adds the action to the internal list and the tensordict. Might be more readable then having yet another wrapper with a `step()`func ^.^